commands/app/migrate.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  addAccountOptions,
+  addConfigOptions,
+  addUseEnvironmentOptions,
+} from '../../lib/commonOpts';
+import {
+  trackCommandMetadataUsage,
+  trackCommandUsage,
+} from '../../lib/usageTracking';
+import { i18n } from '../../lib/lang';
+import { ApiErrorContext, logError } from '../../lib/errorHandlers';
+import { EXIT_CODES } from '../../lib/enums/exitCodes';
+import { getAccountConfig } from '@hubspot/local-dev-lib/config';
+import { ArgumentsCamelCase, Argv, CommandModule } from 'yargs';
+import { MigrateAppOptions } from '../../types/Yargs';
+import { migrateApp2023_2, migrateApp2025_2 } from '../../lib/app/migrate';
+import { PLATFORM_VERSIONS } from '@hubspot/local-dev-lib/constants/projects';
+import { logger } from '@hubspot/local-dev-lib/logger';
+import { uiBetaTag, uiLink } from '../../lib/ui';
+
+const { v2023_2, v2025_2, unstable } = PLATFORM_VERSIONS;
+export const validMigrationTargets = [v2023_2, v2025_2, unstable];
+
+const command = 'migrate';
+const describe = undefined; // uiBetaTag(i18n(`commands.project.subcommands.migrateApp.header.text.describe`), false);
+
+export async function handler(options: ArgumentsCamelCase<MigrateAppOptions>) {
+  const { derivedAccountId, platformVersion } = options;
+  await trackCommandUsage('migrate-app', {}, derivedAccountId);
+  const accountConfig = getAccountConfig(derivedAccountId);
+
+  if (!accountConfig) {
+    logger.error(
+      i18n(`commands.project.subcommands.migrateApp.errors.noAccountConfig`)
+    );
+    return process.exit(EXIT_CODES.ERROR);
+  }
+
+  logger.log('');
+  logger.log(
+    uiBetaTag(
+      i18n(`commands.project.subcommands.migrateApp.header.text`),
+      false
+    )
+  );
+  logger.log(
+    uiLink(
+      i18n(`commands.project.subcommands.migrateApp.header.link`),
+      'https://developers.hubspot.com/docs/platform/migrate-a-public-app-to-projects'
+    )
+  );
+  logger.log('');
+
+  try {
+    if (platformVersion === v2025_2 || platformVersion === unstable) {
+      await migrateApp2025_2(derivedAccountId, options);
+    } else {
+      await migrateApp2023_2(derivedAccountId, options, accountConfig);
+    }
+  } catch (error) {
+    if (
+      error &&
+      typeof error === 'object' &&
+      'errors' in error &&
+      Array.isArray(error.errors)
+    ) {
+      error.errors.forEach(err => logError(err));
+    } else {
+      logError(error, new ApiErrorContext({ accountId: derivedAccountId }));
+    }
+    await trackCommandMetadataUsage(
+      'migrate-app',
+      { successful: false },
+      derivedAccountId
+    );
+    process.exit(EXIT_CODES.ERROR);
+  }
+
+  await trackCommandMetadataUsage(
```

Comment:
I'm not sure what the best way to do this would be, but it might be nice to be able to differentiate between this new migration and the soon-to-be-deprecated one in our tracking

Reply:
Yeah, maybe I could add a tracking to the `hs project migrate-app` handler that is like `migrate-app-deprecated`?  That seems kind of gross, but we can at least subtract those from the `migrate-app` total and get an idea.

Reply:
Yeah something like that could work. You could also call this one `app-migrate` and differentiate that way

Reply:
But the more I think about it. Maybe it's not really important to differentiate. I'm not sure if we'd even need that info

Reply:
Gotcha, I'll leave it alone as is.  Actually in hindsight, we are going to deprecate and delete `hs project migrate-app` so once we release the next major we will only have `hs app migrate`